static/js/common/upload-addon.js

```diff
@@ -429,7 +429,8 @@
                     $("<strong>").text(message).appendTo(upload_results);
 
                     // Specific messages for unlisted addons.
-                    if (isUnlisted()) {
+                    var validation_type = results.validation.detected_type
+                    if (["extension", "dictionary", "languagepack"].indexOf(validation_type) && isUnlisted()) {
```

Comment:
`.indexOf` returns an index or `-1` if not in the list (what's written would test if validation_type wasn't "extension", because that's index=0, so would be true for not in the list or index=1 or index=2).  
`["extension", "dictionary", "languagepack"].indexOf(validation_type) != -1`